consensus-types/light-client/update.go

```diff
@@ -548,8 +644,19 @@ func (u *updateElectra) AttestedHeader() interfaces.LightClientHeader {
 	return u.attestedHeader
 }
 
-func (u *updateElectra) SetAttestedHeader(header interfaces.LightClientHeader) {
+func (u *updateElectra) SetAttestedHeader(header interfaces.LightClientHeader) error {
+	if header.Version() != version.Electra {
+		return fmt.Errorf("header version %s is not %s", version.String(header.Version()), version.String(version.Electra))
+	}
 	u.attestedHeader = header
+
+	proto, ok := header.Proto().(*pb.LightClientHeaderDeneb)
+	if !ok {
+		return fmt.Errorf("header type %T is not %T", proto, &pb.LightClientHeaderDeneb{})
+	}
```

Comment:
Electra update uses Deneb header because there is no change to the header in Electra fork